k8sutils/pkg/workload/workload.go

```diff
@@ -157,3 +158,44 @@ func GetInstrumentationLabelTexts(workloadLabels map[string]string, workloadKind
 
 	return
 }
+
+func GetWorkloadObject(ctx context.Context, name string, kind WorkloadKind, namespace string, kubeClient client.Client) (metav1.Object, error) {
+	switch kind {
+	case WorkloadKindDeployment:
+		var deployment appsv1.Deployment
+		err := kubeClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, &deployment)
+		if err != nil {
+			return nil, err
+		}
+		return &deployment, nil
+
+	case WorkloadKindStatefulSet:
+		var statefulSet appsv1.StatefulSet
+		err := kubeClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, &statefulSet)
+		if err != nil {
+			return nil, err
+		}
+		return &statefulSet, nil
+
+	case WorkloadKindDaemonSet:
+		var daemonSet appsv1.DaemonSet
+		err := kubeClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, &daemonSet)
+		if err != nil {
+			return nil, err
+		}
+		return &daemonSet, nil
+
+	default:
+		return nil, errors.New("failed to get workload object for kind: " + string(kind))
+	}
+}
+
+func ExtractServiceNameFromAnnotations(annotations map[string]string, defaultName string) string {
+	if annotations == nil {
+		return defaultName
+	}
+	if reportedName, exists := annotations[consts.OdigosReportedNameAnnotation]; exists && reportedName != "" {
+		return reportedName
+	}
+	return defaultName
+}
```

Comment:
I remember @blumamir had a code for generating the service name based on the container name as well - to avoid conflicting service names. Do we need to handle this here?